src/eslint/configuration.js

```diff
@@ -91,7 +91,7 @@ const baseRules = {
   "dot-notation": "error",
   "no-redeclare": "error",
   "arrow-parens": "error",
-  "arrow-body-style": ["error", "always"],
+  "arrow-body-style": ["error", "as-needed"],
```

Comment:
I prefer it as "always" because I don't like the special syntax needed for blockless arrow functions. **For very simple functions** (single line) it looks somewhat fine but in other cases it makes the code looks inconsistent with the rest of the code that doesn't need special syntax and special semantics (implicit return).

From my experience those are their downsides

1. It's harder to debug functions without a block. e.g. If I have

```js
foo.bar((baz) => {
  // code
})
```

I can simply

```diff
foo.bar((baz) => {
+  debugger // or console.log(...)
  // code
})
```

But if I have `foo.bar((baz) => foo(baz))` it's more difficult to write a debugger/print statement - I'd have to do a mini-refactor of the code for that and revert it after. Using "bodyless" style makes it more cumbersome to quickly debug something and I don't think the "readabilty" gains are worth it.

2. Object literal looks less readable as showcased by how `getTypescriptOverride` changed because the `{` for opening the object is glued to the `(`.

So in my opinion "bodyless" only looks good in the `.foo(val => short)` expressions and terrible/inconsistent everywhere else, hence why it's as `"always"` right now.

I'll ask for a third opinion here from @mutantcornholio.

If we're going with `as-needed` I'll vote for going with `{ "requireReturnForObjectLiteral": true }` (https://eslint.org/docs/rules/arrow-body-style#requirereturnforobjectliteral)

Reply:
I really miss to see these nice shorthands... aren't they worth adding few curly braces sometimes to debug? )) 
![image](https://user-images.githubusercontent.com/1177472/168824868-185931e8-e356-442b-8bd2-fe8112c0ba12.png)

Reply:
![image](https://user-images.githubusercontent.com/1177472/168825800-a303bd3f-3634-4cc4-aff6-5dce99174da6.png)
![image](https://user-images.githubusercontent.com/1177472/168825814-f2d643bc-d0e3-4d3d-96fe-86ad677cd14a.png)

isn't it too verbose for such straighforward operations?

Reply:
And I would actually prefer `as-needed` without `requireReturnForObjectLiteral` :D  
It's a matter of habit and aesthetics, so I sure can live with other options  

As all options have their pros and cons, should we pick the option with least changes to make?

Reply:
Or wait, that's actually what @mordamax originally suggested, right?

Do we want to vote on `requireReturnForObjectLiteral` separately?

Reply:
I'd summarize so far as

`["error", "always"]`

- In favor: @joao-paulo-parity 
- Against: @mordamax 

`["error", "as-needed"]`

- In favor: @mordamax 
- Against: @joao-paulo-parity 

`["error", "as-needed", { "requireReturnForObjectLiteral": true }]`

- In favor: @joao-paulo-parity

Reply:
Yeah, I'm with @mordamax on second option

Reply:
Given that no one is strictly against `["error", "as-needed", { "requireReturnForObjectLiteral": true }]` unlike the other options, can we go with that?

Reply:
If we're voting, I'm voting for second option. I like writing stuff like 
```ts
sites.map((url, index) => ({
  name: `site${index}`,
  url,
}))
```
and would like to keep doing it.

Reply:
Am  I right, that `{ "requireReturnForObjectLiteral": true }` will still end up with something like this ?
<img width="440" alt="image" src="https://user-images.githubusercontent.com/1177472/168843531-b4e85fe2-7fc2-4af7-a69c-26f037f12ea4.png">

if so, then I'd vote also for `["error", "as-needed"]` w/o requireReturnForObjectLiteral

Reply:
> Am I right, that { "requireReturnForObjectLiteral": true } will still end up with something like this ?

No because that rule only matters for returning objects. Please check out the example in https://eslint.org/docs/rules/arrow-body-style#requirereturnforobjectliteral.

Reply:
> but no one is opposing against it

I **do** oppose against it.  
Not strongly, as it's all codestyle and I could live with any configuration, but that's what we're discussing, right?  
For me, `requireReturnForObjectLiteral` doesn't bring much to the table and forbids me to write the code how I like to write it.  

@mordamax I left an example right above your comment. This is a flag that forbids to quickly return object from an arrow function, and would require a function body with a `return`.

Reply:
as I understand only if you return an object. 
in case of expression like this (attached) it works as short-hand

<img width="747" alt="image" src="https://user-images.githubusercontent.com/1177472/168916290-5357fb19-ba93-4052-9611-b4d59d21ae53.png">

Reply:
Exactly. And I'd like to be able to return an object with a simpler syntax.

Reply:
Even though ideally i'd like `requireReturnForObjectLiteral: false`, I think `as-needed` already good improvement.
So if @joao-paulo-parity is strongly against it, while you're not strongly against it (same as me), I'd end up with `requireReturnForObjectLiteral: true` as a decent compromise for majority of use-cases of short-hand expressions. 
If you agree, lets merge, otherwise, i'd propose to keep "never", as it was before, since it's the way how many projects already have been built